### ISSUE TITLE
Problem compressing image extensions

### ISSUE THREAD

**Issue by aaronroodman** [esheldon/fitsio#34]
Erin,
I have a little script to pull off the 8 F&A CCD's extensions from a raw image, but when I try to also use compression I get an error. 

Here is the script:

import fitsio
hdu = fitsio.FITS("/Users/roodman/Astrophysics/DESData/DECam_00232698.fits.fz")
newhdu = fitsio.FITS('test.fits','rw')
newhdu.write(data=None,header=hdu[0].read_header())
newhdu.write_image(img=hdu['FS1'].read_image(),header=hdu['FS1'].read_header(),extname='FS1',compress='RICE')

Here is the error:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/fitsio/fitslib.py", line 469, in write_image
    self[-1].write_image(img)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/fitsio/fitslib.py", line 1093, in write_image
    self._FITS.write_image(self.ext+1, img_send)
IOError: FITSIO status = 302: column number < 1 or > tfields
Specified column number is out of range: 2
  There are 1 columns in this table.

I have an existing script using pyfits which does work fine, but takes 12 seconds/image, and was hoping that fitsio would be much faster. 

thanks,  Aaron

ps. I am using the version of fitsio from pip.

**Comment by esheldon** [comment]
This sort of thing works for me using master, maybe this was a bug recently fixed.  I used the script below.  Maybe try installing from master for now?  I plan a new release quite soon

Aside 1:  I recommend you use .read() and .write(), that way you don't have to know what type the data is; it will call the correct method for you.

Aside 2: you probably don't want to use the default compression tiling.

```python
import numpy
import fitsio

data=numpy.arange(20*20,dtype='f8').reshape(20,20)

hdr0={'test':3}
hdr1={'blah':'hello'}

with fitsio.FITS('test.fits','rw',clobber=True) as fout:
    fout.write(data=None,header=hdr0)
    fout.write(data,
               header=hdr1,
               extname='FS1',
               compress='RICE')
```

**Comment by aaronroodman** [comment]
ok - I’ll try that - thanks,  Aaron

On Mar 19, 2015, at 2:20 PM, Erin Sheldon <notifications@github.com<mailto:notifications@github.com>> wrote:


This sort of thing works for me using master, maybe this was a bug recently fixed. I used the script below. Maybe try installing from master for now? I plan a new release quite soon

Aside 1: I recommend you use .read() and .write(), that way you don't have to know what type the data is; it will call the correct method for you.

Aside 2: you probably don't want to use the default compression tiling.

import numpy
import fitsio

data=numpy.arange(20*20,dtype='f8').reshape(20,20)

hdr0={'test':3}
hdr1={'blah':'hello'}

with fitsio.FITS('test.fits','rw',clobber=True) as fout:
    fout.write(data=None,header=hdr0)
    fout.write(data,
               header=hdr1,
               extname='FS1',
               compress='RICE')

—
Reply to this email directly or view it on GitHub<https://github.com/esheldon/fitsio/issues/34#issuecomment-83763719>.

_________________________________________________________________________
 Prof. Aaron Roodman
 Chair Dept. of Particle Physics & Astrophysics
 SLAC National Accelerator Laboratory
 Stanford University


 SLAC National Accelerator Laboratory        E-mail: roodman@slac.stanford.edu<mailto:roodman@slac.stanford.edu>
 2575 Sand Hill Rd.                                           Phone:  650-926-2705
 MS 29                                                                  Fax:  650-926-2657
 Menlo Park, CA 94025                                    URL: http://www.slac.stanford.edu/~roodman
_________________________________________________________________________

**Comment by esheldon** [comment]
Hi Aaron -

Was your problem resolved?  I just released 0.9.7 by the way.

**Comment by aaronroodman** [comment]
Actually I don't remember either way, only that  I just stuck with pyfits in the end

Thanks Aaron

Sent from my iPad

On Apr 19, 2015, at 5:30 PM, Erin Sheldon <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Aaron -

Was your problem resolved? I just released 0.9.7 by the way.

-
